src/front/wgsl.rs

```diff
@@ -276,6 +278,21 @@ impl<'a> Lexer<'a> {
             input: &original_input[..used_len],
         })
     }
+
+    fn skip_same_tokens(&mut self, what: &str) -> bool {
```

Comment:
I wonder why we aren't simply skipping the string here, i.e. `self.input.starts_with(what)`

Reply:
I tried just skipping if the same thing, but I think that tokenizing is better because we want to be a little loose with whether spaces between namespace and double colons are allowing, I think. The standard is a little vague, but I think this is what it's saying (https://gpuweb.github.io/gpuweb/wgsl.html#imports).

Reply:
isn't it called on a single namespace anyway? what kind of tokens are we expecting to be skipped here?

Reply:
It is just called on a single namespace, but I wanted `std :: glsl:: distance`  to  parse as well as `std::glsl::distance`.

Reply:
Confirmed by https://matrix.to/#/!MFogdGJfnZLrDmgkBN:matrix.org/$cmtoMc-VHCJTK0BTRxv6zPwMrjnlpz-M9rZAP5x7q8o?via=matrix.org&via=mozilla.org&via=kyju.org.

Reply:
just to be clear on the motivation here, I want the lexer to be as simple as possible. Every new method on the lexer has to be carefully considered. In this case, what you really are looking for is this: given a list of `namespaces: &[&str]`, you want to check if the current stream of tokens corresponds to
```
[
  `Token::ident(namespace[0])`,
  `Token::Separator('::')`,
  `Token::ident(namespace[1])`,
 ... etc
]
```
So it seems to me that this could be implemented in terms of `expect_token` instead of providing a string that is re-parsed again and again.

Reply:
That's a good point, I think you're right. I want to make sure the code for this is easily adaptable to the future when there may be many places something may be imported from, so I'll try doing it this way? Would this involve essentially making a `Vec` of tokens that represent the namespace?

Reply:
please try, I'm sure we can get it to work with tokens